### PR TITLE
api handlers: do not set TlsVerify

### PR DESCRIPTION
**Pull request by vrothberg**
Commit 52a4642edd8a2c2f62d10c2180d785b4f04f18c5 forced the default to
`true` in the code along with the docs.  The problem with hard-setting
it is that it breaks how the optional bool is meant to be used by
containers/image.  If it is not set, c/image will default to true unless
otherwise configured in `/etc/containers/registries.conf`.  If set,
c/image will use the specified value as it assumes the _user_ knows what
they are doing.

In other words: without this fix, the API ignores the `insecure`
settings in `registries.conf`.

Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/api/handlers/libpod/kube.go [hunk 1]
```diff
@@ -29,8 +29,7 @@ func KubePlay(w http.ResponseWriter, r *http.Request) {
 		StaticMACs  []string          `schema:"staticMACs"`
 		NoHosts     bool              `schema:"noHosts"`
 	}{
-		TLSVerify: true,
-		Start:     true,
```
Comment (marshall-lee) [inline pkg/api/handlers/libpod/kube.go:33]: I didn't touch this file though

File: pkg/api/handlers/compat/images_build.go [hunk 2]
```diff
@@ -140,7 +140,6 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		Registry:      "docker.io",
 		Rm:            true,
 		ShmSize:       64 * 1024 * 1024,
-		TLSVerify:     true,
```
Comment (marshall-lee) [inline pkg/api/handlers/compat/images_build.go:143]: Okay now I see that it additionally checks the existence of `"tlsVerify"` query parameter below in the code:

```go
if _, found := r.URL.Query()["tlsVerify"]; found {
	systemContext.DockerInsecureSkipTLSVerify = types.NewOptionalBool(!query.TLSVerify)
	systemContext.OCIInsecureSkipTLSVerify = !query.TLSVerify
	systemContext.DockerDaemonInsecureSkipTLSVerify = !query.TLSVerify
}
```

But does it really break anything? I see, setting a default value just doesn't have any effect cause the parameter only used when it's been actually assigned some value.

Maybe just keep it for documenting purposes? After all it says in docs that it's true by default.

File: pkg/api/handlers/libpod/kube.go (hunk 1 above)
Comment (vrothberg) [reply pkg/api/handlers/libpod/kube.go:33]: Yes, that predates the other PR.

**Comment by marshall-lee** [comment]
@vrothberg 

> In other words: without this fix, the API ignores the insecure settings in registries.conf.

Have you reproduced it?

File: pkg/api/handlers/compat/images_build.go (hunk 2 above)
Comment (vrothberg) [reply pkg/api/handlers/compat/images_build.go:143]: You are absolutely right.  I shouldn't have checked it without a fresh coffee.

` r.URL.Query()["tlsVerify"]` is ignoring the default we set.

> Maybe just keep it for documenting purposes? After all it says in docs that it's true by default.

Yes, I am OK with keeping it as is.  The default is set further down the stack in any case but it doesn't hurt.  As you mentioned, at least for docs.

**Comment by vrothberg** [comment]
Closing as everything's in order.
